packages/core/src/__tests__/acceptance/extension-point.acceptance.ts

```diff
@@ -0,0 +1,131 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  BINDING_METADATA_KEY,
+  Context,
+  Getter,
+  MetadataInspector,
+} from '@loopback/context';
+import {expect} from '@loopback/testlab';
+import {addExtension, extensionPoint, extensions} from '../..';
+
+describe('extension point', () => {
+  describe('@extensionPoint', () => {
+    it('specifies name of the extension point', () => {
+      @extensionPoint('greeters')
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+
+      expect(
+        MetadataInspector.getClassMetadata(
+          BINDING_METADATA_KEY,
+          GreetingService,
+        ),
+      ).to.eql({tags: {name: 'greeters'}});
+    });
+  });
+
+  describe('@extensions', () => {
+    let ctx: Context;
+
+    beforeEach(givenContext);
+
+    it('injects a getter function of extensions', async () => {
+      @extensionPoint('greeters')
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+
+      ctx.bind('greeter-service').toClass(GreetingService);
+      registerGreeters('greeters');
+      const greeterService = await ctx.get<GreetingService>('greeter-service');
+      const greeters = await greeterService.greeters();
+      const languages = greeters.map(greeter => greeter.language);
+      expect(languages).to.containEql('en');
+      expect(languages).to.containEql('zh');
+    });
+
+    it('injects extensions based on the name tag of the extension point binding', async () => {
+      class GreetingService {
+        @extensions()
+        public greeters: Getter<Greeter[]>;
+      }
+      ctx
+        .bind('greeter-service')
+        .toClass(GreetingService)
+        .tag({name: 'greeters'}); // Tag the extension point with a name
+      registerGreeters('greeters');
+      const greeterService = await ctx.get<GreetingService>('greeter-service');
+      const greeters = await greeterService.greeters();
+      const languages = greeters.map(greeter => greeter.language);
+      expect(languages).to.containEql('en');
+      expect(languages).to.containEql('zh');
```

Comment:
These three lines are duplicated a lot:

```ts
const languages = greeters.map(greeter => greeter.language);
expect(languages).to.containEql('en');
expect(languages).to.containEql('zh');
```

Please extract them to a shared helper function accepting a list of expected languages. For example:

```ts
expectGreeterLanguages(greeters, ['en', 'zh']);
```

Implementation wise, please find a way how to merge the two assertions into a single one. At the moment, you are allowing the list of greeter to contain additional greeters, is that intentional? Personally, I would expect an assertion like this:

```ts
const actual = [...languages];
actual.sort()
expect(actual).to.deepEqual(['en', 'zh']);
```